src/lib/util/chdcd.cpp

```diff
@@ -1129,6 +1134,459 @@ chd_error chdcd_parse_cue(const char *tocfname, cdrom_toc &outtoc, chdcd_track_i
 	return CHDERR_NONE;
 }
 
+/*---------------------------------------------------------------------------------------
+    chdcd_is_gdicue - determine if CUE contains Redump multi-CUE format for Dreamcast GDI
+----------------------------------------------------------------------------------------*/
+
+/**
+ * Dreamcast GDI has two images on one disc, SINGLE-DENSITY and HIGH-DENSITY. 
+ *
+ * Redump stores both images in a single .cue with a REM comment separating the images.
+ * This multi-cue format replaces the old flawed .gdi format.
+ *
+ *    http://forum.redump.org/topic/19969/done-sega-dreamcast-multicue-gdi/
+ *
+ * This function looks for strings "REM SINGLE-DENSITY AREA" & "REM HIGH-DENSITY AREA"
+ * indicating the Redump multi-cue format and therefore a Dreamcast GDI disc.
+ */
+
+bool chdcd_is_gdicue(const char *tocfname) 
+{
+	FILE *infile;
+	bool has_rem_singledensity = false;
+	bool has_rem_highdensity = false;
+	std::string path = std::string(tocfname);
+
+	infile = fopen(tocfname, "rt");
+	path = get_file_path(path);
+	if (infile == (FILE *)nullptr)
+	{
+		return false;
+	}
+
+	while (!feof(infile))
+	{
+		fgets(linebuffer, 511, infile);
+
+		/* if EOF didn't hit, keep going */
+		if (!feof(infile))
+		{
+			has_rem_singledensity = has_rem_singledensity || !strncmp(linebuffer, "REM SINGLE-DENSITY AREA", 23);
+			has_rem_highdensity = has_rem_highdensity || !strncmp(linebuffer, "REM HIGH-DENSITY AREA", 21);
+		}
+	}
+
+	fclose(infile);
+
+	return has_rem_singledensity && has_rem_highdensity;
+}
+
+/*-----------------------------------------------------------------
+    chdcd_parse_gdicue - parse a Redump multi-CUE for Dreamcast GDI
+------------------------------------------------------------------*/
+
+/**
+ * @fn  chd_error chdcd_parse_gdicue(const char *tocfname, cdrom_toc &outtoc, chdcd_track_input_info &outinfo)
+ *
+ * @brief   Chdcd parse cue.
+ *
+ * @param   tocfname        The tocfname.
+ * @param [in,out]  outtoc  The outtoc.
+ * @param [in,out]  outinfo The outinfo.
+ *
+ * @return  A chd_error.
+ *
+ * Redump assumes SINGLE-DENSITY starts at 0 LBA and HIGH-DENSITY starts at 45000 LBA. This is *NOT* a generic
+ * multi-cue format its only purpose is Dreamcast GDI dumps. 
```

Comment:
actually, this is generic CUE format, but have added few more custom REM tags

Reply:
Thanks. I've fixed the comment.